### ISSUE TITLE
[BUG] RR2 stop passing requests to workers

### ISSUE THREAD

**Issue by dstrop** [spiral/roadrunner#853]
we are upgrading from RR1 to RR2

after deployment everything works
and after some time (few hours) the roadrunner stops responding

I expected to see this happen:
RR keeps proccessing requests indefinitely

Instead, this happened:
RR stops passing requests to workers

the port is still open and listening to requests
```
# curl --max-time 10 -vvv 127.0.0.1:8080
*   Trying 127.0.0.1:8080...
* Connected to 127.0.0.1 (127.0.0.1) port 8080 (#0)
> GET / HTTP/1.1
> Host: 127.0.0.1:8080
> User-Agent: curl/7.74.0
> Accept: */*
> 
* Operation timed out after 10000 milliseconds with 0 bytes received
* Closing connection 0
curl: (28) Operation timed out after 10000 milliseconds with 0 bytes received
```

workers are sitting idle with no execs and i can list them
```
# /usr/local/bin/rr -c '/etc/roadrunner/.rr.yaml' workers
Workers of [http]:
+---------+-----------+---------+---------+---------+--------------------+
|   PID   |  STATUS   |  EXECS  | MEMORY  |  CPU%   |      CREATED       |
+---------+-----------+---------+---------+---------+--------------------+
|    2379 | ready     |       0 | 28 MB   |    0.01 | 4 minutes ago      |
|    2380 | ready     |       0 | 28 MB   |    0.01 | 4 minutes ago      |
+---------+-----------+---------+---------+---------+--------------------+
```
they are rotated as they reach TTL but live past idle TTL
and calling `reset` just hangs indefinitely i can just see the rotating dots
```
Resetting plugin: [http]  ●∙∙
```
also while restarting i am not even getting the `HTTP plugin got restart request. Restarting...` info log

The version of RR used: 2.5.3
i used binary from docker image spiralscout/roadrunner:2.5.3
in php:8.0.12-cli
  
My `.rr.yaml` configuration is:
```yaml
rpc:
  enable: true
  listen: unix:///etc/roadrunner/rr.sock

server:
  command: "php -d opcache.enable_cli=1 /var/www/html/app/worker.php"
  relay: "pipes"

http:
  address: :8080

  pool:
    num_workers: 2

    supervisor:
      watch_tick: 1s # check every 1s

      # Maximal worker memory usage in megabytes (soft limit). Zero means no limit.
      max_worker_memory: 150 # 150MB
      ttl: 300s # maximum time to live for the worker (soft)
      idle_ttl: 60s # maximum allowed amount of time worker can spend in idle before being removed (for weak db connections, soft)
      exec_ttl: 360s # max_execution_time (brutal)

endure:
  log_level: info

logs:
  mode: production
  level: info

metrics:
  address: :8081
```

Errortrace, Backtrace or Panictrace
there are no errors in logs

**Comment by rustatian** [comment]
@dstrop Only one thing that looks suspicious to me is your phrase:
> they are rotated as they reach TTL but live past idle TTL

TTL is not a hard limit. If the worker has the job and it's in the working state, until it returns the response, RR will wait for it to kill. So, the TTL might be postponed.
But, if the workers are rotated even after a longer timeout, this is not a container issue (not a deadlock).

**Comment by dstrop** [comment]
i am assuming
ttl is time after which the worker will receive the stop command
idle_ttl is time since last worker exec or startup after which it will receive the stop command

if i print `rr workers`
i can see as they reach 5m (ttl is 300s) they are replaced with new workers
but idle ttl is 60s and since they are not receiving any requests even `rr workers` shows 0 execs
i would expect them to be rotated when they reach idle ttl and not live past 60s

the logs i added to php RR library showed the workers are waiting for payload and react to pid and stop commands
but there were no requests received from RR

<details>
  <summary>rr workers</summary>
  
```
# rr -c '/etc/roadrunner/.rr.yaml' workers
Workers of [http]:
+---------+-----------+---------+---------+---------+--------------------+
|   PID   |  STATUS   |  EXECS  | MEMORY  |  CPU%   |      CREATED       |
+---------+-----------+---------+---------+---------+--------------------+
|    1724 | ready     |       0 | 28 MB   |    0.00 | 4 minutes ago      |
|    1725 | ready     |       0 | 28 MB   |    0.01 | 4 minutes ago      |
|    1726 | ready     |       0 | 28 MB   |    0.01 | 4 minutes ago      |
|    1727 | ready     |       0 | 28 MB   |    0.02 | 3 minutes ago      |
|    1742 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1741 | ready     |       0 | 28 MB   |    0.02 | 3 minutes ago      |
|    1728 | ready     |       0 | 28 MB   |    0.02 | 3 minutes ago      |
|    1737 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1736 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1738 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1731 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1733 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1729 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1743 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1739 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1735 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1732 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1734 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1730 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1740 | ready     |       0 | 28 MB   |    0.03 | 3 minutes ago      |
|    1744 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1745 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1746 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1747 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1748 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1749 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1750 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1753 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1751 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
|    1752 | ready     |       0 | 28 MB   |    0.01 | 3 minutes ago      |
+---------+-----------+---------+---------+---------+--------------------+
```
</details>

**Comment by rustatian** [comment]
> i am assuming
ttl is time after which the worker will receive the stop command

It's a time when the worker will be:
1. Marked (not stopped) as TTL-ed if it's currently working on the request. <-- in this case, the worker may outlive the TTL time.
2. Stopped if the worker is in any state except `Working`.

> idle_ttl is time since last worker exec or startup after which it will receive the stop command

Yes, this is the time from the last processed request.

> i can see as they reach 5m (ttl is 300s) they are replaced with new workers

I need a minimum sample to reproduce the issue. Or if you are able to reproduce the problem only in your custom setup - I need a core dump.
1. `ulimit -c unlimited`
2. `GOTRACEBACK=crash ./rr ..`
After you see the problem, you may `killall -SIGSEGV rr` and send me a core dump.

**Comment by dstrop** [comment]
ahh i see now
the idle ttl issue is my misunderstanding and is unrelated to the core issue

for worker to be stoped by idle ttl it has to receive at least one request
if worker doesnt receive any request idle ttl is never triggered and worker is stoped only after reaching ttl

what i did:
i run this with the hello world worker
and sent only 1 request and watched
as the single worker reached idle timeout
but the 3 other workers lived until the TTL
where as i expected to see all 4 to be stoped by idle ttl
<details>
  <summary>.rr.yaml</summary>
  
```yaml
rpc:
  listen: tcp://127.0.0.1:6001

server:
  command: "php worker.php"

http:
  address: "0.0.0.0:8080"

  pool:
    num_workers: 4

    supervisor:
      max_worker_memory: 150
      ttl: 300s
      idle_ttl: 60s
      exec_ttl: 60s

logs:
  level: error
  channels:
    server:
      level: info
```
</details>
<details>
  <summary>logs</summary>
  
```
2021-11-06T19:58:14.624Z	INFO	server      	worker start time: 12.228965759277
2021-11-06T19:58:14.650Z	INFO	server      	worker start time: 12.183904647827
2021-11-06T19:58:14.675Z	INFO	server      	worker start time: 12.048959732056
2021-11-06T19:58:14.721Z	INFO	server      	worker start time: 12.854099273682
2021-11-06T19:59:23.724Z	WARN	server      	worker idle timeout reached	{"pid": 14}
2021-11-06T19:59:23.780Z	INFO	server      	worker start time: 13.642072677612
2021-11-06T20:03:14.724Z	WARN	server      	worker TTL reached	{"pid": 17}
2021-11-06T20:03:14.725Z	WARN	server      	worker TTL reached	{"pid": 18}
2021-11-06T20:03:14.730Z	WARN	server      	worker TTL reached	{"pid": 19}
2021-11-06T20:03:14.758Z	INFO	server      	worker start time: 12.322902679443
2021-11-06T20:03:14.764Z	INFO	server      	worker start time: 14.150857925415
2021-11-06T20:03:14.765Z	INFO	server      	worker start time: 13.571977615356
```
</details>

although this is little troubling as we are establishing connections on worker initialization before receiving first request
but it should not be an issue as we see enough trafic to hopefully prevent this from ever being an issue

as for the core issue i am unable to reproduce it on my machine
i am only able to observe it after deploying in test or prod
where we are running in k8s, so getting a coredump might be a challange
if you need the coredump i can try and figure it out
in the mean time i have RR deployed with --debug so i have can easily use the /debug/pprof

**Comment by rustatian** [comment]
Nevermind, I guess I reproduced the issue, testing the fix...

**Comment by rustatian** [comment]
@dstrop Fixed in the 2.5.4 https://github.com/spiral/roadrunner-binary/releases/tag/v2.5.4

**Comment by dstrop** [comment]
its not yet fixed

the go routine with the Vec lock can still get stuck on https://github.com/spiral/roadrunner/blob/v2.5.1/worker_watcher/container/channel/vec.go#L59
as i can see in pprof goroutine

<details>
<summary>pprof goroutine</summary>

```
goroutine profile: total 388
163 @ 0x4382f6 0x44918c 0x449166 0x464b85 0xb08725 0xb086fb 0xb08f78 0xb13fce 0xb139c9 0xb15535 0xbf8105 0xd776c5 0xd7f7fb 0x6d0c2f 0x6d419b 0x6cfd08 0x468ae1
#	0x464b84	sync.runtime_SemacquireMutex+0x24								runtime/sema.go:71
#	0xb08724	sync.(*RWMutex).RLock+0xa4									sync/rwmutex.go:63
#	0xb086fa	github.com/spiral/roadrunner/v2/worker_watcher/container/channel.(*Vec).Pop+0x7a		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/container/channel/vec.go:102
#	0xb08f77	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Take+0x37			github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:79
#	0xb13fcd	github.com/spiral/roadrunner/v2/pool.(*StaticPool).takeWorker+0x4d				github.com/spiral/roadrunner/v2@v2.5.1/pool/static_pool.go:226
#	0xb139c8	github.com/spiral/roadrunner/v2/pool.(*StaticPool).execWithTTL+0xc8				github.com/spiral/roadrunner/v2@v2.5.1/pool/static_pool.go:178
#	0xb15534	github.com/spiral/roadrunner/v2/pool.(*supervised).Exec+0x94					github.com/spiral/roadrunner/v2@v2.5.1/pool/supervisor_pool.go:59
#	0xbf8104	github.com/spiral/roadrunner-plugins/v2/http/handler.(*Handler).ServeHTTP+0x6a4			github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/handler/handler.go:190
#	0xd776c4	github.com/spiral/roadrunner-plugins/v2/http.(*Plugin).ServeHTTP+0x284				github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/plugin.go:333
#	0xd7f7fa	github.com/spiral/roadrunner-plugins/v2/http/middleware/static.(*Plugin).Middleware.func1+0x57a	github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/middleware/static/plugin.go:150
#	0x6d0c2e	net/http.HandlerFunc.ServeHTTP+0x2e								net/http/server.go:2046
#	0x6d419a	net/http.serverHandler.ServeHTTP+0x43a								net/http/server.go:2878
#	0x6cfd07	net/http.(*conn).serve+0xb07									net/http/server.go:1929

147 @ 0x4382f6 0x44918c 0x449166 0x464b85 0x4729c5 0x473f96 0x473f75 0xb08407 0xb09d38 0xb0968e 0xb0a437 0xb0a7c5 0x468ae1
#	0x464b84	sync.runtime_SemacquireMutex+0x24							runtime/sema.go:71
#	0x4729c4	sync.(*Mutex).lockSlow+0x164								sync/mutex.go:138
#	0x473f95	sync.(*Mutex).Lock+0x35									sync/mutex.go:81
#	0x473f74	sync.(*RWMutex).Lock+0x14								sync/rwmutex.go:111
#	0xb08406	github.com/spiral/roadrunner/v2/worker_watcher/container/channel.(*Vec).Push+0x66	github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/container/channel/vec.go:40
#	0xb09d37	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Release+0x57		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:219
#	0xb0968d	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Allocate+0x3cd		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:192
#	0xb0a436	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).wait+0x1d6		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:295
#	0xb0a7c4	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).addToWatch.func1+0x24	github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:311

30 @ 0x4382f6 0x430973 0x463389 0x49f1b2 0x4a051a 0x4a0508 0x4a9d5e 0x4a9d56 0x49a1d2 0x8d9b5a 0x8d9b46 0x8da725 0x468ae1
#	0x463388	internal/poll.runtime_pollWait+0x88		runtime/netpoll.go:234
#	0x49f1b1	internal/poll.(*pollDesc).wait+0x31		internal/poll/fd_poll_runtime.go:84
#	0x4a0519	internal/poll.(*pollDesc).waitRead+0x259	internal/poll/fd_poll_runtime.go:89
#	0x4a0507	internal/poll.(*FD).Read+0x247			internal/poll/fd_unix.go:167
#	0x4a9d5d	os.(*File).read+0x5d				os/file_posix.go:32
#	0x4a9d55	os.(*File).Read+0x55				os/file.go:119
#	0x49a1d1	io.copyBuffer+0x1b1				io/io.go:423
#	0x8d9b59	io.Copy+0x39					io/io.go:382
#	0x8d9b45	os/exec.(*Cmd).writerDescriptor.func1+0x25	os/exec/exec.go:311
#	0x8da724	os/exec.(*Cmd).Start.func1+0x24			os/exec/exec.go:441

30 @ 0x47f5e5 0x4b0b3c 0x4a9568 0x8da814 0x8da800 0xb07ac9 0xb0707c 0xb0a2a7 0xb0a7c5 0x468ae1
#	0x47f5e4	syscall.Syscall6+0x4									syscall/asm_linux_amd64.s:43
#	0x4b0b3b	os.(*Process).blockUntilWaitable+0x9b							os/wait_waitid.go:33
#	0x4a9567	os.(*Process).wait+0x27									os/exec_unix.go:23
#	0x8da813	os.(*Process).Wait+0x53									os/exec.go:132
#	0x8da7ff	os/exec.(*Cmd).Wait+0x3f								os/exec/exec.go:507
#	0xb07ac8	github.com/spiral/roadrunner/v2/worker.(*Process).Wait+0x28				github.com/spiral/roadrunner/v2@v2.5.1/worker/worker.go:141
#	0xb0707b	github.com/spiral/roadrunner/v2/worker.(*SyncWorkerImpl).Wait+0x1b			github.com/spiral/roadrunner/v2@v2.5.1/worker/sync_worker.go:246
#	0xb0a2a6	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).wait+0x46		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:275
#	0xb0a7c4	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).addToWatch.func1+0x24	github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:311

6 @ 0x4382f6 0x448092 0x9453be 0x468ae1
#	0x9453bd	github.com/spiral/endure/pkg/container.(*Endure).poll.func1+0x9d	github.com/spiral/endure@v1.0.7/pkg/container/poller.go:11

1 @ 0x40b8f4 0x465238 0x975079 0x468ae1
#	0x465237	os/signal.signal_recv+0x97	runtime/sigqueue.go:169
#	0x975078	os/signal.loop+0x18		os/signal/signal_unix.go:24

1 @ 0x4382f6 0x405565 0x40511d 0xb0847e 0xb09d38 0xb0968e 0xb0a437 0xb0a7c5 0x468ae1
#	0xb0847d	github.com/spiral/roadrunner/v2/worker_watcher/container/channel.(*Vec).Push+0xdd	github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/container/channel/vec.go:59
#	0xb09d37	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Release+0x57		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:219
#	0xb0968d	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Allocate+0x3cd		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:192
#	0xb0a436	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).wait+0x1d6		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:295
#	0xb0a7c4	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).addToWatch.func1+0x24	github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:311

1 @ 0x4382f6 0x40640c 0x405e38 0x13f0931 0x468ae1
#	0x13f0930	github.com/spiral/roadrunner-binary/v2/internal/cli/serve.NewCommand.func1.1+0x30	github.com/spiral/roadrunner-binary/v2/internal/cli/serve/command.go:67

1 @ 0x4382f6 0x40640c 0x405e78 0x945873 0x468ae1
#	0x945872	github.com/spiral/endure/pkg/container.(*Endure).startMainThread.func1+0x52	github.com/spiral/endure@v1.0.7/pkg/container/poller.go:42

1 @ 0x4382f6 0x430973 0x463389 0x49f1b2 0x4a051a 0x4a0508 0x5429e9 0x553c25 0x6c9ebf 0x468ae1
#	0x463388	internal/poll.runtime_pollWait+0x88		runtime/netpoll.go:234
#	0x49f1b1	internal/poll.(*pollDesc).wait+0x31		internal/poll/fd_poll_runtime.go:84
#	0x4a0519	internal/poll.(*pollDesc).waitRead+0x259	internal/poll/fd_poll_runtime.go:89
#	0x4a0507	internal/poll.(*FD).Read+0x247			internal/poll/fd_unix.go:167
#	0x5429e8	net.(*netFD).Read+0x28				net/fd_posix.go:56
#	0x553c24	net.(*conn).Read+0x44				net/net.go:183
#	0x6c9ebe	net/http.(*connReader).backgroundRead+0x3e	net/http/server.go:672

1 @ 0x4382f6 0x430973 0x463389 0x49f1b2 0x4a250c 0x4a24f9 0x544195 0x55e668 0x55d83d 0x6d4714 0x6d433d 0x13fc3ed 0x13fc3b5 0x468ae1
#	0x463388	internal/poll.runtime_pollWait+0x88						runtime/netpoll.go:234
#	0x49f1b1	internal/poll.(*pollDesc).wait+0x31						internal/poll/fd_poll_runtime.go:84
#	0x4a250b	internal/poll.(*pollDesc).waitRead+0x22b					internal/poll/fd_poll_runtime.go:89
#	0x4a24f8	internal/poll.(*FD).Accept+0x218						internal/poll/fd_unix.go:402
#	0x544194	net.(*netFD).accept+0x34							net/fd_unix.go:173
#	0x55e667	net.(*TCPListener).accept+0x27							net/tcpsock_posix.go:140
#	0x55d83c	net.(*TCPListener).Accept+0x3c							net/tcpsock.go:262
#	0x6d4713	net/http.(*Server).Serve+0x393							net/http/server.go:3001
#	0x6d433c	net/http.(*Server).ListenAndServe+0x7c						net/http/server.go:2930
#	0x13fc3ec	github.com/spiral/roadrunner-binary/v2/internal/debug.(*Server).Start+0x4c	github.com/spiral/roadrunner-binary/v2/internal/debug/server.go:31
#	0x13fc3b4	github.com/spiral/roadrunner-binary/v2/internal/cli.NewCommand.func1.1+0x14	github.com/spiral/roadrunner-binary/v2/internal/cli/root.go:73

1 @ 0x4382f6 0x430973 0x463389 0x49f1b2 0x4a250c 0x4a24f9 0x544195 0x55e668 0x55d83d 0x6d4714 0xd74ca5 0x468ae1
#	0x463388	internal/poll.runtime_pollWait+0x88					runtime/netpoll.go:234
#	0x49f1b1	internal/poll.(*pollDesc).wait+0x31					internal/poll/fd_poll_runtime.go:84
#	0x4a250b	internal/poll.(*pollDesc).waitRead+0x22b				internal/poll/fd_poll_runtime.go:89
#	0x4a24f8	internal/poll.(*FD).Accept+0x218					internal/poll/fd_unix.go:402
#	0x544194	net.(*netFD).accept+0x34						net/fd_unix.go:173
#	0x55e667	net.(*TCPListener).accept+0x27						net/tcpsock_posix.go:140
#	0x55d83c	net.(*TCPListener).Accept+0x3c						net/tcpsock.go:262
#	0x6d4713	net/http.(*Server).Serve+0x393						net/http/server.go:3001
#	0xd74ca4	github.com/spiral/roadrunner-plugins/v2/http.(*Plugin).serveHTTP+0x204	github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/http.go:24

1 @ 0x4382f6 0x430973 0x463389 0x49f1b2 0x4a250c 0x4a24f9 0x544195 0x56323c 0x561c5d 0x954def 0x468ae1
#	0x463388	internal/poll.runtime_pollWait+0x88					runtime/netpoll.go:234
#	0x49f1b1	internal/poll.(*pollDesc).wait+0x31					internal/poll/fd_poll_runtime.go:84
#	0x4a250b	internal/poll.(*pollDesc).waitRead+0x22b				internal/poll/fd_poll_runtime.go:89
#	0x4a24f8	internal/poll.(*FD).Accept+0x218					internal/poll/fd_unix.go:402
#	0x544194	net.(*netFD).accept+0x34						net/fd_unix.go:173
#	0x56323b	net.(*UnixListener).accept+0x1b						net/unixsock_posix.go:167
#	0x561c5c	net.(*UnixListener).Accept+0x3c						net/unixsock.go:260
#	0x954dee	github.com/spiral/roadrunner-plugins/v2/rpc.(*Plugin).Serve.func1+0x4e	github.com/spiral/roadrunner-plugins/v2@v2.5.2/rpc/plugin.go:89

1 @ 0x4382f6 0x448092 0x13f05fe 0x95deee 0x95e5dc 0x13fc48a 0x13fc48b 0x13fc419 0x437f27 0x468ae1
#	0x13f05fd	github.com/spiral/roadrunner-binary/v2/internal/cli/serve.NewCommand.func1+0x67d	github.com/spiral/roadrunner-binary/v2/internal/cli/serve/command.go:79
#	0x95deed	github.com/spf13/cobra.(*Command).execute+0x60d						github.com/spf13/cobra@v1.2.1/command.go:856
#	0x95e5db	github.com/spf13/cobra.(*Command).ExecuteC+0x3bb					github.com/spf13/cobra@v1.2.1/command.go:974
#	0x13fc489	github.com/spf13/cobra.(*Command).Execute+0x49						github.com/spf13/cobra@v1.2.1/command.go:902
#	0x13fc48a	main.run+0x4a										github.com/spiral/roadrunner-binary/v2/cmd/rr/main.go:22
#	0x13fc418	main.main+0x18										github.com/spiral/roadrunner-binary/v2/cmd/rr/main.go:16
#	0x437f26	runtime.main+0x226									runtime/proc.go:255

1 @ 0x4382f6 0x448092 0xb15a28 0x468ae1
#	0xb15a27	github.com/spiral/roadrunner/v2/pool.(*supervised).Start.func1+0xa7	github.com/spiral/roadrunner/v2@v2.5.1/pool/supervisor_pool.go:89

1 @ 0x4382f6 0x44918c 0x449166 0x464b85 0xb08725 0xb086fb 0xb08f78 0xb13fce 0xb139c9 0xb15535 0xbf8105 0xd776c5 0xd7fb3d 0x6d0c2f 0x6d419b 0x6cfd08 0x468ae1
#	0x464b84	sync.runtime_SemacquireMutex+0x24								runtime/sema.go:71
#	0xb08724	sync.(*RWMutex).RLock+0xa4									sync/rwmutex.go:63
#	0xb086fa	github.com/spiral/roadrunner/v2/worker_watcher/container/channel.(*Vec).Pop+0x7a		github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/container/channel/vec.go:102
#	0xb08f77	github.com/spiral/roadrunner/v2/worker_watcher.(*workerWatcher).Take+0x37			github.com/spiral/roadrunner/v2@v2.5.1/worker_watcher/worker_watcher.go:79
#	0xb13fcd	github.com/spiral/roadrunner/v2/pool.(*StaticPool).takeWorker+0x4d				github.com/spiral/roadrunner/v2@v2.5.1/pool/static_pool.go:226
#	0xb139c8	github.com/spiral/roadrunner/v2/pool.(*StaticPool).execWithTTL+0xc8				github.com/spiral/roadrunner/v2@v2.5.1/pool/static_pool.go:178
#	0xb15534	github.com/spiral/roadrunner/v2/pool.(*supervised).Exec+0x94					github.com/spiral/roadrunner/v2@v2.5.1/pool/supervisor_pool.go:59
#	0xbf8104	github.com/spiral/roadrunner-plugins/v2/http/handler.(*Handler).ServeHTTP+0x6a4			github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/handler/handler.go:190
#	0xd776c4	github.com/spiral/roadrunner-plugins/v2/http.(*Plugin).ServeHTTP+0x284				github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/plugin.go:333
#	0xd7fb3c	github.com/spiral/roadrunner-plugins/v2/http/middleware/static.(*Plugin).Middleware.func1+0x8bc	github.com/spiral/roadrunner-plugins/v2@v2.5.2/http/middleware/static/plugin.go:176
#	0x6d0c2e	net/http.HandlerFunc.ServeHTTP+0x2e								net/http/server.go:2046
#	0x6d419a	net/http.serverHandler.ServeHTTP+0x43a								net/http/server.go:2878
#	0x6cfd07	net/http.(*conn).serve+0xb07									net/http/server.go:1929

1 @ 0x462ea5 0x92beb5 0x92bccd 0x928e4b 0x937a5a 0x93860e 0x6d0c2f 0x6d2529 0x6d419b 0x6cfd08 0x468ae1
#	0x462ea4	runtime/pprof.runtime_goroutineProfileWithLabels+0x24	runtime/mprof.go:746
#	0x92beb4	runtime/pprof.writeRuntimeProfile+0xb4			runtime/pprof/pprof.go:724
#	0x92bccc	runtime/pprof.writeGoroutine+0x4c			runtime/pprof/pprof.go:684
#	0x928e4a	runtime/pprof.(*Profile).WriteTo+0x14a			runtime/pprof/pprof.go:331
#	0x937a59	net/http/pprof.handler.ServeHTTP+0x499			net/http/pprof/pprof.go:253
#	0x93860d	net/http/pprof.Index+0x12d				net/http/pprof/pprof.go:371
#	0x6d0c2e	net/http.HandlerFunc.ServeHTTP+0x2e			net/http/server.go:2046
#	0x6d2528	net/http.(*ServeMux).ServeHTTP+0x148			net/http/server.go:2424
#	0x6d419a	net/http.serverHandler.ServeHTTP+0x43a			net/http/server.go:2878
#	0x6cfd07	net/http.(*conn).serve+0xb07				net/http/server.go:1929
```
</details>

**Comment by rustatian** [comment]
Yeah, forgot the good case. BTW you don't need to send the goroutines every time, it's useless in this case.

**Comment by rustatian** [comment]
@dstrop https://github.com/spiral/roadrunner-binary/releases/tag/v2.5.5, ok, now there are no more blocking sends.
